test/jdk/java/lang/Math/Clamp.java

```diff
@@ -0,0 +1,267 @@
+    }
+
+    private static int testLongClamp() {
+        int failures = 0;
+        long[][] tests = {
+                {0L, 1L, 2L, 1L},
+                {0L, 0L, 2L, 0L},
+                {1L, 0L, 2L, 1L},
+                {2L, 0L, 2L, 2L},
+                {3L, 0L, 2L, 2L},
+                {0L, Long.MIN_VALUE, Long.MAX_VALUE, 0},
+                {Long.MIN_VALUE, Long.MIN_VALUE, Long.MAX_VALUE, Long.MIN_VALUE},
+                {Long.MAX_VALUE, Long.MIN_VALUE, Long.MAX_VALUE, Long.MAX_VALUE},
+                {0, 1, 1, 1},
+                {Long.MAX_VALUE, 1, 1, 1}
+        };
+        long[][] exceptionTests = {
+                {1L, 2L, 0L},
+                {1, Long.MAX_VALUE, Long.MIN_VALUE}
+        };
+        for (long[] test : tests) {
+            long value = test[0];
+            long min = test[1];
+            long max = test[2];
+            long expected = test[3];
+            failures += checkEquals("(long) Math.clamp(" + value + ", " + min + ", " + max + ")", Math.clamp(value, min, max), expected);
+            failures += checkEquals("(long) StrictMath.clamp(" + value + ", " + min + ", " + max + ")", StrictMath.clamp(value, min, max), expected);
+        }
+        for (long[] test : exceptionTests) {
+            long value = test[0];
+            long min = test[1];
+            long max = test[2];
+            failures += checkIllegalArgumentException("(long) Math.clamp(" + value + ", " + min + ", " + max + ")",
+                    () -> Math.clamp(value, min, max));
+            failures += checkIllegalArgumentException("(long) StrictMath.clamp(" + value + ", " + min + ", " + max + ")",
+                    () -> StrictMath.clamp(value, min, max));
+        }
+        return failures;
+    }
+
+    private static int testDoubleClamp() {
+        int failures = 0;
+        double[][] tests = {
+                // value, min, max, expected
+                {-0.1, 0.0, 0.5, 0.0},
+                {-0.0, 0.0, 0.5, 0.0},
+                {0.0, 0.0, 0.5, 0.0},
+                {Double.MIN_VALUE, 0.0, 0.5, Double.MIN_VALUE},
+                {0.2, 0.0, 0.5, 0.2},
+                {Math.nextDown(0.5), 0.0, 0.5, Math.nextDown(0.5)},
+                {0.5, 0.0, 0.5, 0.5},
+                {Math.nextUp(0.5), 0.0, 0.5, 0.5},
+                {0.6, 0.0, 0.5, 0.5},
+
+                {Double.MAX_VALUE, 0.0, Double.POSITIVE_INFINITY, Double.MAX_VALUE},
+                {Double.POSITIVE_INFINITY, 0.0, Double.MAX_VALUE, Double.MAX_VALUE},
+                {-Double.MAX_VALUE, Double.NEGATIVE_INFINITY, 0.0, -Double.MAX_VALUE},
+                {Double.NEGATIVE_INFINITY, -Double.MAX_VALUE, 0.0, -Double.MAX_VALUE},
+
+                {-1.0, -0.0, 0.0, -0.0},
+                {-0.0, -0.0, 0.0, -0.0},
+                {0.0, -0.0, 0.0, 0.0},
+                {1.0, -0.0, 0.0, 0.0},
+                {-1.0, 0.0, 0.0, 0.0},
+                {-0.0, 0.0, 0.0, 0.0},
+                {0.0, 0.0, 0.0, 0.0},
+                {1.0, 0.0, 0.0, 0.0},
```

Comment:
What about adding similar tests for the [-0.0, -0.0] interval?

Reply:
Added, thanks

Reply:
@amaembo I think you added tests for the [0.0, 0.0] interval, rather than [-0.0, -0.0]